paasta_tools/cli/cmds/logs.py

```diff
@@ -549,6 +646,167 @@ def tail_logs(self, service, levels, components, clusters, raw_mode=False):
                 log.warn('Terminating.')
                 break
 
+    def print_logs_by_time(self, service, start_time, end_time, levels, components, clusters, raw_mode):
+        aggregated_logs = []
+
+        def default_fn(scribe_env):
+            stream_name = get_log_name_for_service(service)
```

Comment:
Make a function for calculating the stream name based on service/cluster/etc; this and refactor suggested above would make `default_fn`/`marathon_fn`/etc identical, plus you could reuse the stream name calculation logic between `tail_logs`, `print_logs_by_time`, and `print_last_n_logs`